modules/library/render/src/main/java/org/geotools/renderer/style/MarkFactoryListPredicate.java

```diff
@@ -0,0 +1,36 @@
+/*
+ *    GeoTools - The Open Source Java GIS Toolkit
+ *    http://geotools.org
+ *
+ *    (C) 2021, Open Source Geospatial Foundation (OSGeo)
+ *
+ *    This library is free software; you can redistribute it and/or
+ *    modify it under the terms of the GNU Lesser General Public
+ *    License as published by the Free Software Foundation;
+ *    version 2.1 of the License.
+ *
+ *    This library is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *    Lesser General Public License for more details.
+ */
+package org.geotools.renderer.style;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+/** Mark factory filtering predicate based on a provided class name identifiers list. */
+public class MarkFactoryListPredicate implements Predicate<MarkFactory> {
```

Comment:
Excellent :)

I am not sure there is anything specific about MarkFactory in this class, since it is just checking classname against a list.

You may be able to set this up as  a generic `org.geotools.util.factory.FactoryListPredicate<? extends Factory>` since this is a nice understandable way of filtering.

Reply:
The problem on making the filter class more generic is MarkFactory does not extend Factory or any other interface, so I need to make MarkFactory as type hierarchy root entity.
Also I tried to move the class to gt-metadata but since MarkFactory is a gt-render type the compilation fails on that type resolution.

Reply:
I am sure a generic `org.geotools.util.factory.FactoryListPredicate<T> implements Predicate<T> ` would work fine then.

I just assumed factory because of the name.

Reply:
Does it need to be public?